### DISCUSSION TITLE
STM32WL - 1st microcontroller SoC with LoRa

### DISCUSSION THREAD

**Discussion by Tico06** [micropython/micropython#10807] [STM32 / Pyboard]
Hi All,

I am wondering where we are with STM32WL support.

This thread says it doesn't fit:
[STM32WL - 1st microcontroller SoC with LoRa](https://forum.micropython.org/viewtopic.php?f=12&t=9407&start=10)

But it looks dated, and may be we have progressed here ?

Thanks and regards,

Eric.

**Comment by jimmo** [comment]
It is supported -- there's a board definition for the nucleo and the LoRa driver is almost ready to be merged.

https://github.com/micropython/micropython/tree/master/ports/stm32/boards/NUCLEO_WL55

https://github.com/micropython/micropython/pull/9130

**Comment by Tico06** [comment]
Hi Jimmo,

Thanks for your answer. But what will be left on the flash for the scripts ? Looks challenging isn't it ?

I'll order a card and give it a try.

What is missing to have this fork merged ?

Thanks and regards

Eric.

**Reply by jimmo** [reply]
> Thanks for your answer. But what will be left on the flash for the scripts ? Looks challenging isn't it ?

Yes, it doesn't have a huge amount on on-chip flash (256kiB) but there's currently 232-213=19kiB free of firmware space plus 24kiB for the fileysystem. There's likely lots of firmware features you could disable to free up more space if necessary (e.g. you could disable the on-board compiler... we're working on making it possible to do a REPL without an onboard compiler too).

You'd freeze the LoRa driver into the firmware (and any libraries), so then the full 24kiB filesystem would be available for Python code.

**Reply by DuaneKaufman** [reply]
Dear All,

Has anyone produced a Micropython image for the NUCLEO_WL55?

Sincerely,
Duane

**Reply by andrewleech** [reply]
As @jimmo [said above](https://github.com/orgs/micropython/discussions/10807#discussioncomment-5072592) it's already defined as a standard board in the stm port. This means the images can be downloaded on https://micropython.org/download/?mcu=stm32wl

**Reply by davefes** [reply]
Do the images also have `micropython-lora` in them?

Should these images run on a EByte E77-400M22S https://www.cdebyte.com/products/E77-400M22S
without too much drama?

Thanks.

**Reply by jimmo** [reply]
@davefes I don't know what `micropython-lora` is, but this board is supported by https://github.com/micropython/micropython-lib/tree/master/micropython/lora
(It's not frozen by default into the firmware, you need to install it with `mip` for now. I've just sent https://github.com/micropython/micropython/pull/12909 to add it to the firmware by default)

**Reply by davefes** [reply]
Yes, I meant `mircopython/lora'.  
Thank you.

**Comment by DuaneKaufman** [comment]
Great! Thanks On Nov 7, 2023, at 4:51 PM, Andrew Leech ***@***.***> wrote:﻿
As @jimmo said above it's already defined as a standard board in the stm port. This means the images can be downloaded on https://micropython.org/download/?mcu=stm32wl
